### PR TITLE
Enable watchOS for Remote Config and ABTesting

### PR DESCRIPTION
**Pull request by chliangGoogle**
ABTesting is a hard dependency so enable it as well.
Testing: I'm able to get config value without any changes using the watchOS sample app.

### CODE REVIEW

**Review by paulb777 — COMMENTED** [review]
Thanks!

Please update CI with a build only `pod lib lint` at https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/remoteconfig.yml#L52

File: README.md
Comment (paulb777) [inline README.md:264]: Also ABTesting at line 256 and its CI

File: Example/watchOSSample/SampleWatchAppWatchKitExtension/ExtensionDelegate.swift
Comment (ryanwilson) [inline Example/watchOSSample/SampleWatchAppWatchKitExtension/ExtensionDelegate.swift:39]: Nit: for those that aren't using a project with a `test` key, can you please change the force unwrap to a safe fallback? Something like `?? "No value for key 'test'"` or just `?? "nil"`

**Review by ryanwilson — COMMENTED** [review]
Awesome! LGTM, please wait for @karenyz approval.

File: FirebaseABTesting.podspec
Comment (paulb777) [inline FirebaseABTesting.podspec:24]: nit: using Ruby variables for the platform target versions is more robust. See https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseCore.podspec#L21

**Review by paulb777 — APPROVED** [review]

File: Example/watchOSSample/SampleWatchAppWatchKitExtension/ExtensionDelegate.swift
Comment (karenyz) [inline Example/watchOSSample/SampleWatchAppWatchKitExtension/ExtensionDelegate.swift:39]: might be helpful if we also print the fetch source?

**Review by karenyz — APPROVED** [review]

**Comment by karenyz** [comment]
Thanks for testing this! 🎉

File: Example/watchOSSample/SampleWatchAppWatchKitExtension/ExtensionDelegate.swift
Comment (chliangGoogle) [reply Example/watchOSSample/SampleWatchAppWatchKitExtension/ExtensionDelegate.swift:39]: Done.

Comment (chliangGoogle) [reply Example/watchOSSample/SampleWatchAppWatchKitExtension/ExtensionDelegate.swift:39]: Done. I specify it in the output that if the config is setup on console.

File: README.md
Comment (chliangGoogle) [reply README.md:264]: Done.

File: FirebaseABTesting.podspec
Comment (chliangGoogle) [reply FirebaseABTesting.podspec]: Done. Also updated remote config.

**Comment by chliangGoogle** [comment]
my clang-format somehow is not in sync even tho it's the same version.

**Comment by paulb777** [comment]
The log might be confusing, since it's actually swiftformat that checks the format of Swift files - not clang-format

**Comment by chliangGoogle** [comment]
Right, I also upgraded swiftformat and I ran both swiftformat and ./script/style.sh. None of them is accepted by CI.

**Comment by paulb777** [comment]
swiftformat should be controlled by the Mintfile and running 0.45.5 both locally and in CI

**Pull request merged**
